### PR TITLE
feat: 🍰 Community Contribution Site And Form

### LINKED ISSUE #2038: 🚀 [Feature] Community contribution site and form
## 🚀 Feature
Create the Community Contributions page 
Create the basic form for community contributions

 
Meine Community_Schreiben

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest
community startpage created, entry form for contributions, list of own contributions and of all contributions of a community prepared. 

erstellte beiträge landen im adminbereich um bestätigt zu werden.

### Issues
- fixes #2038 

### TODO
- editing must not be possible for submitted contributions in the admin area.

### CODE REVIEW

**Comment by ogerly** [comment]
image

**Comment by BerndHueckstaedt** [comment]
Looks good :-)
I would put the menu item "Community" above the profile
I still have a few small text corrections. If you give me the text, I'll make the corrections.

Sieht gut aus :-)
Den Menüpunkt "Gemeinschaft" würde ich über das Profil setzen
Ich habe noch ein paar kleine Text-Korrekturen. Wenn Du mir den Text gibst, mache ich die Korrekturen.

File: frontend/src/components/Contributions/ContributionForm.vue [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        class="text-right"
+        :class="form.memo.length < minlength ? 'text-danger' : 'text-success'"
+      >
+        <span v-if="form.memo.length < minlength">{{ $t('math.equalTo') }} {{ minlength }}</span>
+        <span v-else>{{ $t('math.divide') }} {{ maxlength }}</span>
+      </div>
+      <label class="mt-3">{{ $t('form.amount') }}</label>
+      <b-input-group size="lg" prepend="GDD" append=".00">
+        <b-form-input
+          id="testamount"
```
Comment (Mogge) [inline frontend/src/components/Contributions/ContributionForm.vue:48]:
```suggestion
          id="contribution-amount"
```

File: frontend/src/components/Contributions/ContributionForm.vue [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+ *
+ */
+export default {
+  name: 'ContributionForm',
+  data() {
+    return {
+      minlength: 50,
+      maxlength: 255,
+      lastMonth: new Date(new Date(new Date().setMonth(new Date().getMonth() - 1))),
+      min: new Date(new Date(new Date().setMonth(new Date().getMonth() - 1)).setDate(1)),
```
[the remark this answers is not in the document]
Comment (Mogge) [reply frontend/src/components/Contributions/ContributionForm.vue:77]: For both. `lastMonth` and `min`

File: frontend/src/locales/de.json [hunk 3]
```diff
@@ -26,8 +26,22 @@
...
+    "writing": "Schreiben"
+  },
+  "contribution": {
+    "activity": "Tätigkeit",
+    "formText": {
+      "h3": "Dein Beitrag zum Gemeinwohl",
+      "lastMonth": "Für <b>{month} {year}</b> kannst du noch <b>{creation}</b> GDD einreichen.",
+      "text1": "Bring dich mit deinen Talenten in die Gemeinschaft ein! Dein freiwilliges Engagement honorieren wir mit 20 GDD pro Stunde bis maximal 1.000 GDD im Monat.",
+      "text2": "Beschreibe deine Gemeinwohl-Tätigkeit mit Angabe der Stunden und trage einen Betrag von 20 GDD pro Stunde ein! Nach Bestätigung durch einen Moderator wird der Betrag deinem Konto gutgeschrieben.",
+      "thisMonth": "Für <b>{month} {year}</b> kannst du noch <b>{creation}</b> GDD einreichen. "
```
Comment (Mogge) [inline frontend/src/locales/de.json:41]: What is the difference to `lastMonth`

File: frontend/src/components/Contributions/ContributionForm.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/components/Contributions/ContributionForm.vue:84]: lastMonth() and min() is now computed

File: frontend/src/components/Contributions/ContributionForm.vue [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+      this.$emit('set-contribution', this.form)
+      this.$refs.form.reset()
+      this.form.date = ''
+    },
+  },
+  computed: {
+    disable() {
+      if (
+        this.form.memo.length < this.minlength ||
+        this.form.amount === 0 ||
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/components/Contributions/ContributionForm.vue:104]: ok

File: frontend/src/components/Contributions/ContributionForm.vue [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  },
+  methods: {
+    submit() {
+      this.$emit('set-contribution', this.form)
+      this.$refs.form.reset()
+      this.form.date = ''
+    },
+  },
+  computed: {
+    disable() {
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/components/Contributions/ContributionForm.vue:101]: ok

File: frontend/src/components/Contributions/ContributionForm.vue [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+          id="testamount"
+          v-model="form.amount"
+          type="number"
+          min="1"
+          max="1000"
+        ></b-form-input>
+      </b-input-group>
+
+      <div class="mt-3 text-right">
+        <b-button class="test-submit" type="submit" variant="primary" :disabled="disable">
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/components/Contributions/ContributionForm.vue:57]: ok

File: frontend/src/components/Contributions/ContributionForm.vue [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+        :max="max"
+        :min="min"
+        class="mb-4"
+        reset-value=""
+        :label-no-date-selected="$t('contribution.noDateSelected')"
+        required
+      ></b-form-datepicker>
+      <label class="mt-3">{{ $t('contribution.activity') }}</label>
+      <b-form-textarea
+        id="testmemo"
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/components/Contributions/ContributionForm.vue:30]: ok

File: frontend/src/components/Contributions/ContributionForm.vue (hunk 1 above)
Comment (ogerly) [reply frontend/src/components/Contributions/ContributionForm.vue:48]: ok

File: frontend/src/components/Contributions/ContributionForm.vue [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+      </ul>
+
+      <div class="my-3">
+        <b>{{ $t('contribution.formText.text2') }}</b>
+      </div>
+    </div>
+    <b-form ref="form" @submit.prevent="submit">
+      <label>{{ $t('time.month') }}</label>
+      <b-form-datepicker
+        id="testdate"
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/components/Contributions/ContributionForm.vue:18]: ok

File: frontend/src/locales/de.json (hunk 3 above)
Comment (ogerly) [reply frontend/src/locales/de.json:41]: lastMonth is the full date of new Date() - 1 month, this is the sentence that is then placed on the form.  Here I only need the month written out as text.

File: frontend/src/components/Contributions/ContributionForm.vue [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+        this.form.amount === 0 ||
+        this.form.amount > 1000
+      )
+        return true
+      return false
+    },
+    lastMonthObject() {
+      // new Date().getMonth === 1 If the current month is January, then one year must be gone back in the previous month
+      const obj = {
+        month: new Date(this.lastMonth).toLocaleString(this.$i18n.locale, { month: 'long' }),
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/components/Contributions/ContributionForm.vue:113]: is now this.$d

File: frontend/src/i18n.js [hunk 10]
```diff
@@ -75,6 +75,15 @@ const dateTimeFormats = {
...
     },
+    monthShort: {
+      month: 'short',
+    },
+    month: {
+      month: 'long',
+    },
+    year: {
+      year: 'numeric',
+    },
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/i18n.js:86]: ok

File: frontend/src/components/Contributions/ContributionListItem.vue [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+    },
+    amount: {
+      type: String,
+    },
+    memo: {
+      type: String,
+    },
+    firstName: {
+      type: String,
+      require: false,
```
Comment (Mogge) [inline frontend/src/components/Contributions/ContributionListItem.vue:48]:
```suggestion
      required: false,
```

File: frontend/src/pages/Community.vue [hunk 12]
```diff
@@ -0,0 +1,203 @@
...
+  created() {
+    this.updateListContributions({
+      currentPage: this.currentPage,
+      pageSize: this.pageSize,
+    })
+    this.updateListAllContributions({
+      currentPage: this.currentPage,
+      pageSize: this.pageSize,
+    })
+    this.updateTransactions(0)
```
Comment (Mogge) [inline frontend/src/pages/Community.vue:200]: Why is this needed?

File: frontend/src/routes/router.test.js [hunk 13]
```diff
@@ -50,7 +50,7 @@ describe('router', () => {
       })
 
       it('has sixteen routes defined', () => {
-        expect(routes).toHaveLength(16)
+        expect(routes).toHaveLength(17)
```
Comment (Mogge) [inline frontend/src/routes/router.test.js:53]: Test the new route?

File: frontend/src/locales/en.json [hunk 14]
```diff
@@ -26,8 +26,23 @@
...
     "other-communities": "Other communities",
-    "switch-to-this-community": "Switch to this community"
+    "switch-to-this-community": "Switch to this community",
+    "writing": "Writing"
+  },
+  "contribution": {
+    "activity": "Activity",
+    "formText": {
+      "h3": "Your contribution to the common good",
+      "lastMonth": "For  <b>{monthAndYear}</b>, you can still submit <b>{creation}</b> GDD.",
```
Comment (Mogge) [inline frontend/src/locales/en.json:38]: Is this needed? Looks the same as `thisMonth`

**Review by Mogge — CHANGES_REQUESTED** [review]
Do you want me to write the tests?

File: frontend/src/pages/Community.vue (hunk 12 above)
Comment (ogerly) [reply frontend/src/pages/Community.vue:200]: Yes, we need that at this point, unfortunately, because otherwise the GDD amount is not displayed when I call up the page alone or reload it.

File: frontend/src/routes/router.test.js (hunk 13 above)
Comment (ogerly) [reply frontend/src/routes/router.test.js:53]: i have tested now

File: frontend/src/locales/en.json (hunk 14 above)
Comment (ogerly) [reply frontend/src/locales/en.json:38]: thisMonth is deleted an lastMonth is new **openAmountForMonth**

File: frontend/src/components/Contributions/ContributionListItem.vue [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+    date() {
+      if (this.deletedAt) return this.deletedAt
+      if (this.confirmedAt) return this.confirmedAt
+      return this.contributionDate
+    },
+  },
+  methods: {
+    deleteContribution(id) {
+      this.boxOne = ''
+      this.$bvModal.msgBoxConfirm('Delete Contribution! Are you sure?').then((value) => {
```
Comment (Mogge) [inline frontend/src/components/Contributions/ContributionListItem.vue:99]: This message must be localized.

File: frontend/src/pages/Community.vue [hunk 16]
```diff
@@ -0,0 +1,218 @@
...
+          mutation: updateContribution,
+          variables: {
+            contributionId: data.id,
+            creationDate: data.date,
+            memo: data.memo,
+            amount: data.amount,
+          },
+        })
+        .then((result) => {
+          this.toastSuccess(result.data)
```
Comment (Mogge) [inline frontend/src/pages/Community.vue:104]: Please toast a message and not the result.

File: frontend/src/pages/Community.vue [hunk 17]
```diff
@@ -0,0 +1,218 @@
...
+          fetchPolicy: 'no-cache',
+          mutation: createContribution,
+          variables: {
+            creationDate: data.date,
+            memo: data.memo,
+            amount: data.amount,
+          },
+        })
+        .then((result) => {
+          this.toastSuccess(result.data)
```
Comment (Mogge) [inline frontend/src/pages/Community.vue:80]: Please toast a message and not the result.

File: frontend/src/pages/Community.vue [hunk 18]
```diff
@@ -0,0 +1,218 @@
...
+        .mutate({
+          fetchPolicy: 'no-cache',
+          mutation: deleteContribution,
+          variables: {
+            id: id,
+          },
+        })
+        .then((result) => {
+          // console.log('result', result.data)
+          this.toastSuccess(result.data)
```
Comment (Mogge) [inline frontend/src/pages/Community.vue:126]: Please toast a message and not the result.

**Review by Mogge — DISMISSED** [review]
So gut wie fertig!

File: frontend/src/components/Contributions/ContributionListItem.vue (hunk 15 above)
Comment (ogerly) [reply frontend/src/components/Contributions/ContributionListItem.vue:99]: ok

File: frontend/src/pages/Community.vue (hunk 16 above)
Comment (ogerly) [reply frontend/src/pages/Community.vue:104]: ok

File: frontend/src/pages/Community.vue (hunk 17 above)
Comment (ogerly) [reply frontend/src/pages/Community.vue:80]: ok

File: frontend/src/pages/Community.vue (hunk 18 above)
Comment (ogerly) [reply frontend/src/pages/Community.vue:126]: ok

**Comment by ogerly** [comment]
> 

yes, the two days are still work ... that's what I wrote. subtleties and such ...  :)

File: frontend/src/components/Contributions/ContributionForm.spec.js [hunk 19]
```diff
@@ -0,0 +1,49 @@
...
+      wrapper = Wrapper()
+    })
+
+    it('has a DIV .contribution-form', () => {
+      expect(wrapper.find('div.contribution-form').exists()).toBe(true)
+    })
+
+    it('is submit button disable of true', () => {
+      expect(wrapper.find('button[type="submit"]').attributes('disabled')).toBe('disabled')
+    })
```
Comment (Tirokk) [inline frontend/src/components/Contributions/ContributionForm.spec.js:47]: Does it makes sense to click on the submit button and see what happens?

May we have to fill in some values?

– see that it gets enabled?
- then click submit and see that the right things happen?

File: frontend/src/locales/de.json [hunk 20]
```diff
@@ -26,8 +26,34 @@
     "community": "Gemeinschaft",
     "continue-to-registration": "Weiter zur Registrierung",
     "current-community": "Aktuelle Gemeinschaft",
+    "myContributions": "Meine Beiträge",
     "other-communities": "Weitere Gemeinschaften",
-    "switch-to-this-community": "zu dieser Gemeinschaft wechseln"
+    "switch-to-this-community": "zu dieser Gemeinschaft wechseln",
+    "writing": "Schreiben"
```
Comment (Tirokk) [inline frontend/src/locales/de.json:32]:
```suggestion
    "submitContribution": "Beitrag einreichen"
```

File: frontend/src/locales/en.json [hunk 21]
```diff
@@ -26,8 +26,34 @@
     "community": "Community",
     "continue-to-registration": "Continue to registration",
     "current-community": "Current community",
+    "myContributions": "My contributions",
     "other-communities": "Other communities",
-    "switch-to-this-community": "Switch to this community"
+    "switch-to-this-community": "Switch to this community",
+    "writing": "Writing"
```
Comment (Tirokk) [inline frontend/src/locales/en.json:32]:
```suggestion
    "submitContribution": "Submit contribution"
```

**Review by Tirokk — CHANGES_REQUESTED** [review]
I have encountered some misterios things by manual tryout …

`bibi@bloxberg.de` can still contribute 1000 GDDs in July, though she already got 1000 GDDs in July:

Bildschirmfoto 2022-07-22 um 06 59 47

Bildschirmfoto 2022-07-22 um 06 59 54

Successfully confirmed an additional 500 GDDs by `peter@lustig.de`:

Bildschirmfoto 2022-07-22 um 06 53 19

## Behaviour

### Submit Or Write Contribution

If I enter no date the submit button is enabled:

Bildschirmfoto 2022-07-22 um 07 18 49

### Submit Or Write Contribution – After An Admin Has Deleted One Of Me

If I submitted a contribution which gots deleted by an admin my amount what I can contribute is still lowered by the amount of my deleted one.

## Namings

### Tabs

Bildschirmfoto 2022-07-22 um 07 08 52

- better maybe:
  - DE: "writing": "Schreiben" -->  "submitContribution": "Beitrag einreichen" see suggestion 👇🏼
  - EN: see suggestion 👇🏼

File: frontend/src/locales/de.json [hunk 22]
```diff
@@ -26,8 +26,34 @@
...
+      "t2": "Es wird bald an dieser Stelle die Möglichkeit geben das ein Dialog zwischen Moderatoren und dir stattfinden kann. Solltest du jetzt Probleme haben bitte nimm Kontakt mit dem Support auf."
+    },
+    "delete": "Beitrag löschen! Bist du sicher?",
+    "deleted": "Der Beitrag wurde gelöscht! Wird aber sichtbar bleiben.",
+    "formText": {
+      "h3": "Dein Beitrag zum Gemeinwohl",
+      "maxGDDforMonth": "Du kannst für den ausgewählten Monat nur noch maximal {amount} GDD einreichen.",
+      "openAmountForMonth": "Für <b>{monthAndYear}</b> kannst du noch <b>{creation}</b> GDD einreichen.",
+      "text1": "Bring dich mit deinen Talenten in die Gemeinschaft ein! Dein freiwilliges Engagement honorieren wir mit 20 GDD pro Stunde bis maximal 1.000 GDD im Monat.",
+      "text2": "Beschreibe deine Gemeinwohl-Tätigkeit mit Angabe der Stunden und trage einen Betrag von 20 GDD pro Stunde ein! Nach Bestätigung durch einen Moderator wird der Betrag deinem Konto gutgeschrieben."
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/locales/de.json]: ok

File: frontend/src/locales/de.json [hunk 23]
```diff
@@ -26,8 +26,34 @@
...
+      "li1": "Eingereicht und wartet auf Bestätigung",
+      "li2": "bestätigt",
+      "li3": "gelöscht",
+      "t1": "Hier findest du chronologisch aufgelistet alle deine eingereichten Beiträge. Es gibt drei Darstellungsarten. Du kannst deine Beiträge, welche noch nicht bestätigt wurden, jederzeit bearbeiten.",
+      "t2": "Es wird bald an dieser Stelle die Möglichkeit geben das ein Dialog zwischen Moderatoren und dir stattfinden kann. Solltest du jetzt Probleme haben bitte nimm Kontakt mit dem Support auf."
+    },
+    "delete": "Beitrag löschen! Bist du sicher?",
+    "deleted": "Der Beitrag wurde gelöscht! Wird aber sichtbar bleiben.",
+    "formText": {
+      "h3": "Dein Beitrag zum Gemeinwohl",
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/locales/de.json]: ok

File: frontend/src/components/Contributions/ContributionForm.vue [hunk 24]
```diff
@@ -0,0 +1,166 @@
...
+    minimalDate() {
+      return new Date(this.maximalDate.getFullYear(), this.maximalDate.getMonth() - 1, 1)
+    },
+    disabled() {
+      if (
+        this.form.memo.length < this.minlength ||
+        this.form.amount <= 0 ||
+        this.form.amount > 1000 ||
+        (this.isThisMonth && parseInt(this.form.amount) > parseInt(this.maxGddThisMonth)) ||
+        (!this.isThisMonth && parseInt(this.form.amount) > parseInt(this.maxGddlastMonth))
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/components/Contributions/ContributionForm.vue]: ok

**Review by Tirokk — CHANGES_REQUESTED** [review]
Again some changes to suggest

File: frontend/src/components/Menu/Sidebar.spec.js [hunk 25]
```diff
@@ -47,18 +47,21 @@ describe('Sidebar', () => {
       it('has first nav-item "navigation.transactions" in navbar', () => {
         expect(wrapper.findAll('.nav-item').at(2).text()).toEqual('navigation.transactions')
       })
+      it('has a link to the community area', () => {
+        expect(wrapper.findAll('.nav-item').at(3).text()).toContain('navigation.community')
+      })
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/components/Menu/Sidebar.spec.js]: ok

File: frontend/src/locales/de.json [hunk 26]
```diff
@@ -176,7 +202,9 @@
   "login": "Anmeldung",
   "math": {
     "aprox": "~",
+    "divide": "/",
     "equal": "=",
+    "equalTo": "<",
```
Comment (Tirokk) [inline frontend/src/locales/de.json:207]:
```suggestion
    "lower": "<",
```

Change all usages as well !!!

File: frontend/src/locales/en.json [hunk 27]
```diff
@@ -176,7 +202,9 @@
   "login": "Login",
   "math": {
     "aprox": "~",
+    "divide": "/",
     "equal": "=",
+    "equalTo": "<",
```
Comment (Tirokk) [inline frontend/src/locales/en.json:207]:
```suggestion
    "lower": "<",
```

Change all usages as well !!!

File: frontend/src/components/Contributions/ContributionListItem.vue [hunk 28]
```diff
@@ -0,0 +1,109 @@
...
+  },
+  methods: {
+    deleteContribution(item) {
+      this.$bvModal
+        .msgBoxConfirm(this.$t('contribution.delete'))
+        .then((value) => {
+          this.$emit('delete-contribution', item)
+        })
+        .catch((_err) => {
+          // An error occurred
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/components/Contributions/ContributionListItem.vue]: ok

**Review by Tirokk — CHANGES_REQUESTED** [review]
Two suggestions again …

File: frontend/src/pages/Community.spec.js [hunk 29]
```diff
@@ -0,0 +1,388 @@
...
+        })
+
+        it('toasts the error message', () => {
+          expect(toastErrorSpy).toBeCalledWith('Oh No!')
+        })
+      })
+    })
+
+    describe('delete contribution', () => {
+      describe('with success', () => {
```
Comment (Tirokk) [inline frontend/src/pages/Community.spec.js:299]:
```suggestion
    describe('delete contribution', () => {
      let contributionListComponent

      beforeEach(async () => {
        await wrapper.setData({ tabIndex: 1 })
        contributionListComponent = await wrapper.findComponent({ name: 'ContributionList' })
      })

      describe('with success', () => {
```

File: frontend/src/pages/Community.spec.js [hunk 30]
```diff
@@ -0,0 +1,388 @@
...
+
+      describe('with error', () => {
+        beforeEach(async () => {
+          jest.clearAllMocks()
+          apolloMutationMock.mockRejectedValue({
+            message: 'Oh my god!',
+          })
+          await wrapper
+            .findComponent({ name: 'ContributionForm' })
+            .vm.$emit('delete-contribution', 2)
```
Comment (Tirokk) [inline frontend/src/pages/Community.spec.js:353]:
```suggestion
          contributionListComponent.vm.$emit('delete-contribution', { id: 2 })
```

File: frontend/src/pages/Community.vue [hunk 31]
```diff
@@ -0,0 +1,264 @@
...
+        .query({
+          query: verifyLogin,
+          fetchPolicy: 'network-only',
+        })
+        .then((result) => {
+          const {
+            data: { verifyLogin },
+          } = result
+          this.$store.dispatch('login', verifyLogin)
+        })
```
Comment (Tirokk) [inline frontend/src/pages/Community.vue:233]: Where is the update of:

- `this.items`
- `this.itemsAll`

?
 If I delete my contribution the list in the community tab gets not updated except I reload.

File: frontend/src/pages/Community.vue [hunk 32]
```diff
@@ -0,0 +1,264 @@
...
+      this.form.amount = item.amount
+      this.updateAmount = item.amount
+      this.tabIndex = 0
+    },
+    updateTransactions(pagination) {
+      this.$emit('update-transactions', pagination)
+    },
+  },
+  created() {
+    // verifyLogin is important at this point so that creation is updated if they are deleted in a session in the admin area.
```
Comment (Tirokk) [inline frontend/src/pages/Community.vue:251]:
```suggestion
    // verifyLogin is important at this point so that creation is updated on reload if they are deleted in a session in the admin area.
```

**Review by Tirokk — CHANGES_REQUESTED** [review]
I have still some changes to suggest …

File: frontend/src/locales/de.json (hunk 26 above)
Comment (Tirokk) [reply frontend/src/locales/de.json]: Done

File: frontend/src/components/Contributions/ContributionListItem.spec.js [hunk 33]
```diff
@@ -0,0 +1,157 @@
...
+      })
+
+      describe('confirm deletion', () => {
+        beforeEach(() => {
+          spy = jest.spyOn(wrapper.vm.$bvModal, 'msgBoxConfirm')
+          spy.mockImplementation(() => Promise.resolve(true))
+          wrapper.findAll('div.pointer').at(1).trigger('click')
+        })
+
+        it('opens the modal', () => {
```
[the remark this answers is not in the document]
Comment (Tirokk) [reply frontend/src/components/Contributions/ContributionListItem.spec.js:134]: Works now

**Comment by Mogge** [comment]
> I have encountered some misterios things by manual tryout …
> 
> `bibi@bloxberg.de` can still contribute 1000 GDDs in July, though she already got 1000 GDDs in July:
> 
> Bildschirmfoto 2022-07-22 um 06 59 47
> 
> Bildschirmfoto 2022-07-22 um 06 59 54
> 
> Successfully confirmed an additional 500 GDDs by `peter@lustig.de`:
> 
> Bildschirmfoto 2022-07-22 um 06 53 19
> ## Behaviour
> ### Submit Or Write Contribution
> 
> If I enter no date the submit button is enabled:
> 
> Bildschirmfoto 2022-07-22 um 07 18 49
> ### Submit Or Write Contribution – After An Admin Has Deleted One Of Me
> 
> If I submitted a contribution which gots deleted by an admin my amount what I can contribute is still lowered by the amount of my deleted one.
> ## Namings
> ### Tabs
> 
> Bildschirmfoto 2022-07-22 um 07 08 52
> 
>     * better maybe:
>       
>       * DE: "writing": "Schreiben" -->  "submitContribution": "Beitrag einreichen" see suggestion 👇🏼
>       * EN: see suggestion 👇🏼

File: frontend/src/pages/Community.vue (hunk 31 above)
Comment (Tirokk) [reply frontend/src/pages/Community.vue:245]: done

**Review by Tirokk — APPROVED** [review]
So I approve and do a follow up issue for the little remainders 🚀🚀🚀💫💫💫

File: frontend/src/components/Contributions/ContributionForm.spec.js (hunk 19 above)
Comment (Tirokk) [reply frontend/src/components/Contributions/ContributionForm.spec.js:47]: Done
